documentation/library_guide/parallel_api/range_based_api.rst

```diff
@@ -1,15 +1,107 @@
+
+
+Range-Based API (The standard C++ Ranges Library)
+-------------------------------------------------
+
+The following C++ standard random access adaptors and factories are supported with the oneDPL parallel algorithms:
+
+* ``std::ranges::views::all``: A range adaptor object that returns a view that includes all elements of its range argument.
+* ``std::ranges::iota_view``: A range factory that generates a sequence of N elements, which starts from an initial value and ends by final N-1.
+* ``std::ranges::single_view``: A view that contains exactly one element of a specified value.
+* ``std::ranges::subrange``: A utility that combines together an iterator and a sentinel into a single object that models a view.
+* ``std::ranges::transform_view``: A range adapter that represents a view of a underlying sequence after applying a transformation to each element.
+* ``std::ranges::reverse_view``: A range adapter that produces a reversed sequence of elements provided by another view.
+* ``std::ranges::take_view``: A range adapter that produces a view of the first N elements from another view.
+* ``std::ranges::drop_view``: A range adapter that produces a view excluding the first N elements from another view.
+
+The following algorithms are available in the oneapi::dpl::ext::ranges namespace to use with the mentioned standard ranges:
+
+* ``for_each``
+* ``transform``
+* ``find``
+* ``find_if``
+* ``find_if_not``
+* ``all_of``
+* ``any_of``
+* ``none_of``
+* ``adjacent_find``
+* ``search``
+* ``search_n``
+* ``count_if``
+* ``count``
+* ``equal``
+* ``sort``
+* ``stable_sort``
+* ``is_sorted``
+* ``min_element``
+* ``max_element``
+* ``copy``
+* ``copy_if``
+* ``merge``
+
+Example of Range-Based API Usage (The Standard C++ Ranges Library)
+------------------------------------------------------------------
+
+.. code:: cpp
+
+    using namespace oneapi::dpl::ext::ranges;
+
+    {        
+        std::vector<int> vec_in = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+        std::vector<int> vec_out(vec_in.size());
+
+        auto view_in = std::ranges::views::all(vec_in) | std::ranges::views::reverse;
+        copy(oneapi::dpl::execution::par, view_in, vec_out);
+    }
+    {
+        using shared_allocator = sycl::usm_allocator<int, sycl::usm::alloc::shared>;
+
+        std::vector<int, shared_allocator> vec_in = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+        std::vector<int, shared_allocator> vec_out(vec_in.size());
+
+        auto view_in = std::ranges::subrange(vec_in.begin(), vec_in.end()) | std::ranges::views::reverse;
+        copy(oneapi::dpl::execution::dpcpp_default, view_in, std::span(vec_out));
+    }
+
+Experimental Range-Based API (oneDPL Ranges)
+------------------------------------------------
+
+The following viewable ranges are declared in the ``oneapi::dpl::experimental::ranges`` namespace.
+Only the ranges shown below and ``sycl::buffer`` are available as ranges for range-based algorithms.
```

Comment:
I think we need a bit more description of the experimental range based API and how that differs / co-exists with the productized range base API.  We should mention that it is available with C++17.  Also, it may be good to give some guidance to users about the purpose (current and historically) of this experimental API, and possibly our intentions for the experimental API going forward.

Reply:
I attempted to address this in the reworked page for the experimental range API.